lms/djangoapps/course_home_api/outline/tests/test_goals.py

```diff
@@ -126,3 +129,41 @@ def test_add_invalid_goal(self):
         response = self.save_course_goal('notnumber', False)
         assert response.status_code == 400
         assert len(CourseGoal.objects.filter(user=self.user, course_key=self.course.id)) == 0
+
+
+class TestUnsubscribeAPI(BaseCourseHomeTests):
+    """
+    Testing the unsubscribe API.
+    """
+    def unsubscribe(self, token):
+        url = reverse('course-home:unsubscribe-from-course-goal', kwargs={'token': token})
+        return self.client.post(url)
+
+    def make_goal(self, course_key, **kwargs) -> CourseGoal:
+        return CourseGoal.objects.create(user=self.user, course_key=course_key, **kwargs)
+
+    def test_happy_path(self):
+        goal = self.make_goal(self.course.id, subscribed_to_reminders=True)
+        goal2 = self.make_goal('course-v1:foo+bar+2T2020', subscribed_to_reminders=True)  # a control group
+
+        def unsubscribe_and_check():
+            response = self.unsubscribe(goal.unsubscribe_token)
+            goal.refresh_from_db()
+            goal2.refresh_from_db()
+            assert response.status_code == 200
+            assert not goal.subscribed_to_reminders
+            assert goal2.subscribed_to_reminders
```

Comment:
I realize we might not be adding any other tests, but it feels like maybe this could exist as a higher level function on the class rather than a sub-method in happy path. I suppose it could be moved later too. This is all a nit.

Reply:
This one felt so specific to this test (checking goal2 for example), I figured just keep it inside.